src/app/modules/sgid/sgid.adapter.ts

```diff
@@ -83,28 +131,61 @@ export class SGIDMyInfoData
    * @returns the formatted field.
    */
   _formatFieldValue(attr: ExternalAttr): string | undefined {
-    return this.#payload[attr]
+    console.log('sgid payload attr: ', this.#payload[attr])
+    switch (attr) {
+      case ExternalAttr.RegisteredAddress:
+        return formatAddress(this.#payload[attr])
+      case ExternalAttr.VehicleNo:
+        return formatVehicles(this.#payload[attr])
+      default:
+        return this.#payload[attr]
+    }
   }
 
   /**
-   * Refer to the myInfo data catalogue to see which fields should be read-only
-   * and which fields should be editable by the user.
+   * SGID only returns verified MyInfo fields, unless the field contains marriage-related information
+   * (decision by SNDGO & MSF due to overseas unregistered marriages).
+   * An empty myInfo field will always evaluate
+   * to false so that the field can be filled by form-filler.
+   * SGID returns 'NA' for field values that do not exist (vs empty string returned by Singpass MyInfo)
+   *
+   * The affected marriage fields are:
+   * - marital
+   * - marriagedate
+   * - divorcedate
+   * - countryofmarriage
+   * - marriagecertno
+   *
    * @param attr sgID MyInfo OAuth scope.
    * @param fieldValue FormSG field value.
    * @returns Whether the data/field should be readonly.
    */
   _isDataReadOnly(attr: ExternalAttr, fieldValue: string | undefined): boolean {
     const data = this.#payload[attr]
-    if (!data || !fieldValue) return false
+    if (!data || !fieldValue || fieldValue === 'NA' || data === 'NA')
+      return false
 
     switch (attr) {
-      case ExternalAttr.MobileNumber:
+      case ExternalAttr.MobileNoWithPrefix:
       case ExternalAttr.RegisteredAddress:
       case ExternalAttr.Name:
       case ExternalAttr.PassportNumber:
       case ExternalAttr.DateOfBirth:
       case ExternalAttr.PassportExpiryDate:
+      case ExternalAttr.Sex:
+      case ExternalAttr.Race:
+      case ExternalAttr.Nationality:
+      case ExternalAttr.HousingType:
+      case ExternalAttr.HdbType:
+      case ExternalAttr.BirthCountry:
+      case ExternalAttr.VehicleNo:
+      case ExternalAttr.Employment:
+      case ExternalAttr.WorkpassStatus:
+      case ExternalAttr.WorkpassExpiryDate:
         return !!data
+      // Fields required to always be editable according to MyInfo docs
+      case ExternalAttr.MaritalStatus:
+        return false
       // Fall back to leaving field editable as data shape is unknown.
       default:
         return false
```

Comment:
Outside of your PR but recommend to have error logging here, as this method is just directly called in `getFieldValueForAttr`. In case some of the attributes name got changed etc. and we suddenly keep returning `readOnly: false` we will at least have some logs